prog/vm/github_runner.rb

```diff
@@ -223,6 +223,33 @@ def setup_info
       UBICLOUD_CACHE_URL=#{Config.base_url}/runtime/github/" | sudo tee -a /etc/environment
     COMMAND
 
+    if vm.vm_host.id == Vm[Config.docker_mirror_server_vm_id]&.vm_host&.id
+      mirror_server_vm = Vm[Config.docker_mirror_server_vm_id]
+      mirror_server_address = mirror_server_vm.load_balancer.hostname
+      mirror_server_port = 5000
+      command += <<~COMMAND
+        # Configure Docker daemon with registry mirror
+        if [ -f /etc/docker/daemon.json ] && [ -s /etc/docker/daemon.json ]; then
+          jq '. + {"registry-mirrors": ["https://#{mirror_server_address}:#{mirror_server_port}"]}' /etc/docker/daemon.json | sudo tee /etc/docker/daemon.json.tmp
```

Comment:
We run `mv` and `tee` with `sudo`, so I assumed that accessing `/etc/docker/daemon.json` requires root permissions. Should we also add `sudo` to the `jq` command? Can it access the file without it?

Reply:
Yes `jq` can read the file but there is no harm to do it with sudo, let me change it to not confuse.